gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/services/SubscriptionCacheService.java

```diff
@@ -271,4 +280,16 @@ String buildClientIdCacheKey(Subscription subscription) {
     String buildCacheKeyFromClientInfo(String api, String clientIdOrCertificate, String plan) {
         return String.format("%s.%s.%s", api, clientIdOrCertificate, plan);
     }
+
+    private Set<String> extractApiServersId(Subscription subscription) {
+        final ReactableApi<?> reactableApi = apiManager.get(subscription.getApi());
+        final Set<String> servers;
+        if (reactableApi instanceof Api api) {
+            servers = api.getDefinition().getListeners().stream().flatMap(l -> l.getServers().stream()).collect(Collectors.toSet());
+        } else {
+            servers = Set.of();
+            log.debug("V2 APIs do not support subscription using client certificate");
```

Comment:
Can the reactableApi be `null`? If so the log is talking about v2 API and could be wrong.